### DISCUSSION TITLE
Expand list of discovered spaces with valuable information

### DISCUSSION THREAD

**Discussion by haroldvandekamp** [WikiTransformationProject/wikitraccs-releases#59] [Ideas]
When running the discovery of spaces, the store in SharePoint provides a great list of all the spaces that are found. Can the list be extended with columns providing more valuable information, e.g.:
- Total pages
- Total blogs 
- Total files
- Storage size
- Total users
- Space creation date
- Last activity 
- Space owner(s)
- Visitor count
- Last time page added
- Last time page edited
- Last time blog added
- Last time blog edited
- Total plug-ins / macro's 
- list of plug-ins / macro's used 
- Related category (because spaces can be categorized, making it easier to digest which Hub Sites to create in SharePoint and connect the space to it)

I know this list is too comprehensive ;)
However, these additional information would help deciding wether, when and how to migrate a space. It also helps to make a raw estimate how long the migration of a space would take and to prioritize specific spaces for migration.

**Comment by heinrich-ulbricht** [comment]
@haroldvandekamp This is a good list and I appreciate the feedback very much!

Some of those infos could be gathered pretty fast, like "last time page added". Those are candidates for just adding them because WikiTraccs has this information after getting the list of pages for each space that is scheduled to be migrated to SharePoint.

There is other information that is harder to get. Files and storage size require scanning all attachments of all pages. Don't know if this is even possible without downloading them. Same for the list of macros. I know that there are faster ways to get this information, like looking into the Confluence database, exporting Confluence spaces via out of the box space export and analyzing the XML, or looking at the file system. Here I am not sure if the use case is compelling, because getting this information would be slower via WikiTraccs. On the upside: you won't need to fight with IT departments over privileged access and WikiTraccs would be your one-stop-shop for getting all info, although slower.

Getting this info would require adding a pre-migration scan feature to WikiTraccs that gets information without migrating anything from Confluence to SharePoint.

All in all this would be a valuable medium-term addition to WikiTraccs.
